src/Headless/Avalonia.Headless.Vnc/HeadlessVncPlatformExtensions.cs

```diff
@@ -16,10 +19,14 @@ public static class HeadlessVncPlatformExtensions
         public static int StartWithHeadlessVncPlatform(
             this AppBuilder builder,
             string host, int port,
-            string[] args, ShutdownMode shutdownMode = ShutdownMode.OnLastWindowClose)
+            string[] args,
+            ShutdownMode shutdownMode = ShutdownMode.OnLastWindowClose,
+            string? password = null,
+            ILogger? logger = null)
```

Comment:
ILogger here is a leaking abstraction. Avalonia itself doesn't reference `Microsoft.Extensions.Logging` in any way, and instead we use `Avalonia.Logging.Logger.TryGet()`. You possibly can write an internal ILogger wrapper here. 
And call `Logger.TryGet(LogLevel, LogArea.VncPlatform)?.Log()` from the wrapper.